nntrainer/layers/bn_layer.cpp

```diff
@@ -177,16 +175,15 @@ BatchNormalizationLayer::backwarding(sharedConstTensors derivative,
   dx_2.subtract_i(deviation.divide(cvar).multiply(
     deviation.multiply(deriv).sum(axes_to_reduce)));
 
-  Tensor dx = dx_2.multiply(dx_1);
+  Tensor &dx = net_input[0]->grad;
+  dx = dx_2.multiply(dx_1);
   dx.divide_i(N);
```

Comment:
is it ok to modify this in-place?
also where this grad is initialized?

Reply:
net_input[0]->grad is the return derivative and will be overwritten. No need to initialized. I don't see any problems here. Let me know if it is.

Reply:
Ignore my comment, this is correct.
I was confused.